test/conformance/tests/sriov_operator.go

```diff
@@ -930,13 +939,51 @@ var _ = Describe("[sriov] operator", func() {
 				It("Should be able to create pods successfully if PF is down.Pods are able to communicate with each other on the same node", func() {
 					resourceName := "testresource"
 					sriovNetworkName := "sriovnetwork"
-					testNode := sriovInfos.Nodes[0]
-					sriovDeviceList, err := sriovInfos.FindSriovDevices(testNode)
-					Expect(err).ToNot(HaveOccurred())
-					unusedSriovDevices, err := findUnusedSriovDevices(testNode, sriovDeviceList)
-					defer changeNodeInterfaceState(testNode, unusedSriovDevices[0].Name, true)
-					Expect(err).ToNot(HaveOccurred())
-					createSriovPolicy(unusedSriovDevices[0].Name, testNode, 2, resourceName)
+					var testNode string
+					var unusedSriovDevice *sriovv1.InterfaceExt
+
+					if discovery.Enabled() {
+						var numVfs int
+						var err error
+						testNode, resourceName, numVfs, unusedSriovDevice, err = discovery.DiscoveredResources(clients,
+							sriovInfos.Nodes, operatorNamespace,
+							func(policy sriovv1.SriovNetworkNodePolicy) bool {
+								if policy.Spec.DeviceType != "netdevice" {
+									return false
+								}
+								return true
+
+							},
+							func(node corev1.Node) (bool, *sriovv1.InterfaceExt) {
+								sriovDeviceList, err := sriovInfos.FindSriovDevices(node.ObjectMeta.Name)
```

Comment:
How about fetching the device list into DiscoveredResources and passing it to `filterDevices`? It would make the function more compact and also will stick better with the function name